repos/system_upgrade/el7toel8/actors/biosdevname/actor.py

```diff
@@ -29,9 +29,12 @@ def is_vendor_dell(self):
         context = pyudev.Context()
 
         # There should be only one dmi/id device
-        dmi = pyudev.Enumerator(context).match_subsystem('dmi')
-        dev = list(filter(lambda d: d.sys_name == 'id', dmi))[0]
-        vendor = dev.attributes.get('sys_vendor')
+        dmi = pyudev.Enumerator(context).match_subsystem('dmi').match_sys_name('id')
+        try:
```

Comment:
```
if not dmi:
    return False
```
?

Reply:
Unfortunately this doesn't work. I tried to figure out some more elegant way than using try/except block, but I failed. This also doesn't work.

Reply:
Well I was writing it on a mobile.
I rather expect something like this to work:

```python
dmi = pyudev.Enumerator(context).match_subsystem('dmi').match_sys_name('id')
dev = next(dmi, None)
if not dev:
    return None
vendor = dev.attributes.get('sys_vendor')
return re.search('Dell.*', str(vendor)) is not None
```

Reply:
`dev = next(dmi, None)` won't work either, it would have to be like this:
`dev = next(dmi.__iter__(), None)`, which is really ugly. But the for loop should work, that's the first thing I should have tried...